pkg/services/sqlstore/apikey_test.go

```diff
@@ -27,5 +29,91 @@ func TestApiKeyDataAccess(t *testing.T) {
 			})
 
 		})
+
+		t.Run("Add non expiring key", func(t *testing.T) {
+			cmd := m.AddApiKeyCommand{OrgId: 1, Name: "non-expiring", Key: "asd1", SecondsToLive: 0}
+			err := AddApiKey(&cmd)
+			assert.Nil(t, err)
+
+			query := m.GetApiKeyByNameQuery{KeyName: "non-expiring", OrgId: 1}
+			err = GetApiKeyByName(&query)
+			assert.Nil(t, err)
+
+			assert.Nil(t, query.Result.Expires)
+		})
+
+		t.Run("Add an expiring key", func(t *testing.T) {
+			//expires in one hour
+			cmd := m.AddApiKeyCommand{OrgId: 1, Name: "expiring-in-an-hour", Key: "asd2", SecondsToLive: 3600}
+			err := AddApiKey(&cmd)
+			assert.Nil(t, err)
+
+			query := m.GetApiKeyByNameQuery{KeyName: "expiring-in-an-hour", OrgId: 1}
+			err = GetApiKeyByName(&query)
+			assert.Nil(t, err)
+
+			assert.True(t, *query.Result.Expires >= timeNow().Unix())
+
+			// timeNow() has been called twice since creation; once by AddApiKey and once by GetApiKeyByName
+			// therefore two seconds should be subtracted by next value retuned by timeNow()
+			// that equals the number by which timeSeed has been advanced
+			then := timeNow().Add(-2 * time.Second)
+			expected := then.Add(1 * time.Hour).UTC().Unix()
+			assert.Equal(t, *query.Result.Expires, expected)
+		})
+
+		t.Run("Add a key with invalid lifespan", func(t *testing.T) {
+			//expires in one day
+			cmd := m.AddApiKeyCommand{OrgId: 1, Name: "key-with-invalid-lifespan", Key: "asd3", SecondsToLive: -3600}
```

Comment:
Shouldn't this error out if the value is invalid? If it doesn't right now what will happen? Create a key with expiration in the past?

Reply:
No, it creates a key that does not expire. So it actually ignores the invalid value for SecondsToLive.

Reply:
Ah ok right, small nitpick then, maybe we could change the test name to something like "Add key with negative lifespan". I am usually expecting some error if something is invalid.

Reply:
maybe it's better to return error in that case

Reply:
Don't have all the context here, but one way we've handles this elsewhere is returning a typed error and then if validation error returned we can in the api make sure to return a 400 error instead of 500 error (for unhandled/run time error). I think it's more explicit to return an error here and notify the user.

How is this handled in UI? Can you input a negative timespan? As a general rule UI and API should provide same experience and easiest is that UI respects any error messages returned from the api and adds input validation on top, if needed, to guide the user and provide a better user experience.

Reply:
> How is this handled in UI? Can you input a negative timespan?

No, the UI does not allow this.

Reply:
Then the API should return an error as well.